models/models.go

```diff
@@ -0,0 +1,136 @@
+package models
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"sync"
+	"github.com/gobuffalo/envy"
+	"crypto/tls"
+	"net"
+	"strings"
+)
+
+var MONGOD_URL = envy.Get("MONGOD_URL","localhost:27017")
+var ON_ATLAS = envy.Get("ON_ATLAS", "false")
+
+var (
+	once       sync.Once
+	dbInstance DB
+)
+
+type DB struct {
+	session *mgo.Session
+	logger  *log.Entry
+}
+
+type ReadRequest struct {
+	dBName         string
+	collectionName string
+	logger         *log.Entry
+}
+
+type WriteRequest struct {
+	DbName         string
+	CollectionName string
+	Data           interface{}
+}
+
+func (db *DB) Write(wR *WriteRequest) {
+	collection := db.session.DB(wR.DbName).C(wR.CollectionName) // maybe can cache this later?
+	err := collection.Insert(wR.Data)
+	if err != nil {
+		log.Fatal("failed to insert data into db. dbName: %v, collectionName: %v, data: %v, error: %v", wR.DbName, wR.CollectionName, wR.Data, err)
+	}
+}
+
+func (db *DB) CreateCappedCollection(dbName string, collectionName string, capacity int) {
+	collectionInfo := &mgo.CollectionInfo{
+		Capped:   true,
+		MaxBytes: capacity,
+	}
+	err := db.session.DB(dbName).C(collectionName).Create(collectionInfo)
+	if err != nil {
+		log.WithField("Function", "CreateCappedCollection").Error(err)
+	}
+}
+
+func (db *DB) GetUserById(id string) (User, error) {
+	collection := db.session.DB("auth").C("users")
```

Comment:
Is it more "go" to name this 'c' instead of 'collection'?

Reply:
yeah go takes its naming conventions from c so it makes sense to do that. Im just a java boy so i have a bad naming habit ill fix it. ;)